nzedb/Categorize.php

```diff
@@ -1362,95 +1524,109 @@ public function isMusic()
 		}
 	}
 
+	/**
+	 * @return bool
+	 */
 	public function isMusicForeign()
 	{
-		if ($this->categorizeForeign) {
-			if (preg_match('/[ \-\._](brazilian|chinese|croatian|danish|deutsch|dutch|estonian|flemish|finnish|french|german|greek|hebrew|icelandic|italian|ita|latin|mandarin|nordic|norwegian|polish|portuguese|japenese|japanese|russian|serbian|slovenian|spanish|spanisch|swedish|thai|turkish|bl|cz|de|es|fr|ger|heb|hu|hun|it(a| 19|20\d\d)|jap|ko|kor|nl|pl|se)[ \-\._]/i', $this->releaseName)) {
-				$this->tmpCat = Category::MUSIC_FOREIGN;
+		if ($this->categorizeForeign && preg_match('/[ \-\._](brazilian|chinese|croatian|danish|deutsch|dutch|estonian|flemish|finnish|french|german|greek|hebrew|icelandic|italian|ita|latin|mandarin|nordic|norwegian|polish|portuguese|japenese|japanese|russian|serbian|slovenian|spanish|spanisch|swedish|thai|turkish|bl|cz|de|es|fr|ger|heb|hu|hun|it(a| 19|20\d\d)|jap|ko|kor|nl|pl|se)[ \-\._]/i', $this->releaseName)) {
+			$this->tmpCat = Category::MUSIC_FOREIGN;
 
-				return true;
-			}
+			return true;
 		}
 
 		return false;
 	}
 
+	/**
+	 * @return bool
+	 */
 	public function isAudiobook()
 	{
-		if ($this->categorizeForeign) {
-			if (preg_match('/Audiobook/i', $this->releaseName)) {
-				$this->tmpCat = Category::MUSIC_FOREIGN;
+		if ($this->categorizeForeign && stripos($this->releaseName, 'Audiobook') !== false) {
+			$this->tmpCat = Category::MUSIC_FOREIGN;
 
-				return true;
-			}
+			return true;
 		}
 
 		return false;
 	}
 
+	/**
+	 * @return bool
+	 */
 	public function isMusicVideo()
 	{
 		if (preg_match('/(720P|x264)\-(19|20)\d\d\-[a-z0-9]{1,12}/i', $this->releaseName)) {
 			if ($this->isMusicForeign()) {
-				return true;
-			} else {
-				$this->tmpCat = Category::MUSIC_VIDEO;
-
 				return true;
 			}
+
+			$this->tmpCat = Category::MUSIC_VIDEO;
+
+			return true;
 		}
 		if (preg_match('/[a-z0-9]{1,12}\-(19|20)\d\d\-(720P|x264)/i', $this->releaseName)) {
 			if ($this->isMusicForeign()) {
-				return true;
-			} else {
-				$this->tmpCat = Category::MUSIC_VIDEO;
-
 				return true;
 			}
+
+			$this->tmpCat = Category::MUSIC_VIDEO;
+
+			return true;
 		}
 
 		return false;
 	}
 
+	/**
+	 * @return bool
+	 */
 	public function isMusicLossless()
 	{
 		if (preg_match('/\[(19|20)\d\d\][-._ ]\[FLAC\]|(\(|\[)flac(\)|\])|FLAC\-(19|20)\d\d\-[a-z0-9]{1,12}|\.flac"|(19|20)\d\d\sFLAC|[-._ ]FLAC.+(19|20)\d\d[-._ ]| FLAC$/i', $this->releaseName)) {
 			if ($this->isMusicForeign()) {
-				return true;
```

Comment:
Please revert changes to this method, as it needs to be refactored to remove the excessive returns.